scripts/setup/upgrade-postgres

```diff
@@ -43,12 +38,27 @@ pg_ctlcluster "$UPGRADE_FROM" main stop
     rm -rf "$TEMP_CONF_DIR"
 )
 
-pg_upgradecluster "$UPGRADE_FROM" main
+# Capture the output so we know where the path to the post-upgrade scripts is
+UPGRADE_LOG=$(mktemp)
+pg_upgradecluster "$UPGRADE_FROM" main --method=upgrade --link | tee "$UPGRADE_LOG"
+SCRIPTS_PATH=$(grep -o '/var/log/postgresql/pg_upgradecluster-10-12-main.*' "$UPGRADE_LOG")
+rm "$UPGRADE_LOG"
 
+# If the upgrade completed successfully, lock in the new version in
+# our configuration immediately
 crudini --set /etc/zulip/zulip.conf postgresql version "$UPGRADE_TO"
+
+# Update the statistics
+su postgres -c "$SCRIPTS_PATH/analyze_new_cluster.sh"
+
+# Start the database up cleanly
 "$ZULIP_PATH"/scripts/zulip-puppet-apply -f
 
+# XXX Why?
```

Comment:
@timabbott: Do you recall what this step is necessary for?  None of the _data_ in the database is changed, so I'm not clear what we need it for.

Reply:
Good question.  I don't think a memcached restart should be required.  If we'd upgraded the Python major version, it could change the format of pickled objects or whatnot, but whenever we do that, there's a service restart anyway (which will, in turn, change the prefix used for all cache keys).

Probably useless and should be removed.

Reply:
This got here because it's a step in the previous upgrade documentation.  It looks like it first showed up in the first upgrade steps, in 5d2984fb461ce2c828e61e00075ef65641a2ba25.

I'll drop it from here -- should we drop it from the Trusty -> Xenial and Stretch -> Buster documentation as well?

Reply:
Those change the major Python and thus do have more of a reason to do it... actually, thinking about this more clearly, the fact that we do `upgrade-zulip-stage-2` does not flush memcached or change the key prefix (since we're reusing an existing deployment), so I think it **is** important to do this whenever we're changing the major Python version.

Reply:
Which means it's not important for `upgrade-postgres`, but is important as part of the upgrade steps.  Rather than leave it here as fodder for confusion in the future, I'll add a commit which moves it to the doc steps -- I don't think one more command there will be problematic.

Reply:
Actually, `do-release-upgrade` both installs a new version of `memcached` and also forces a server restart.  I think we're guaranteed that memcached is already empty.

Reply:
Well, in theory the `memcached` version could have not changed over 2 years, but I suppose we know for a fact that it did for these specific cases.  And in any case, `scripts/setup/flush-memcached` would be the correct tool to use.